nixos/modules/services/networking/frr.nix

```diff
@@ -25,210 +69,260 @@ let
+          #
+          #MAX_FDS=1024
+
+          # Uncomment this option if you want to run FRR as a non-root user. Note that
+          # you should know what you are doing since most of the daemons need root
+          # to work. This could be useful if you want to run FRR in a container
+          # for instance.
+          # FRR_NO_ROOT="yes"
+
+          # For any daemon, you can specify a "wrap" command to start instead of starting
+          # the daemon directly. This will simply be prepended to the daemon invocation.
+          # These variables have the form daemon_wrap, where 'daemon' is the name of the
+          # daemon (the same pattern as the daemon_options variables).
+          #
+          # Note that when daemons are started, they are told to daemonize with the `-d`
+          # option. This has several implications. For one, the init script expects that
+          # when it invokes a daemon, the invocation returns immediately. If you add a
+          # wrap command here, it must comply with this expectation and daemonize as
+          # well, or the init script will never return. Furthermore, because daemons are
+          # themselves daemonized with -d, you must ensure that your wrapper command is
+          # capable of following child processes after a fork() if you need it to do so.
+          #
+          # If your desired wrapper does not support daemonization, you can wrap it with
+          # a utility program that daemonizes programs, such as 'daemonize'. An example
+          # of this might look like:
+          #
+          # bgpd_wrap="/usr/bin/daemonize /usr/bin/mywrapper"
+          #
+          # This is particularly useful for programs which record processes but lack
+          # daemonization options, such as perf and rr.
+          #
+          # If you wish to wrap all daemons in the same way, you may set the "all_wrap"
+          # variable.
+          #
+          #all_wrap=""
+        '';
+      };
 
-  meta.maintainers = with lib.maintainers; [ woffs ];
+      systemd.tmpfiles.rules = [ "d /run/frr 0750 frr frr -" ];
+
+      systemd.services.frr = {
+        description = "FRRouting";
+        documentation = [ "https://frrouting.readthedocs.io/en/latest/setup.html" ];
+        wants = [ "network.target" ];
+        after = [
+          "network-pre.target"
+          "systemd-sysctl.service"
+        ];
+        before = [ "network.target" ];
+        # onFailure = [ "heartbeat-failed@%n" ];
+        wantedBy = [ "multi-user.target" ];
+        startLimitIntervalSec = 180;
+        reloadIfChanged = true;
+        restartTriggers = [
+          configFile
+          daemonList
+        ];
+        serviceConfig = {
+          Nice = -5;
+          Type = "forking";
+          NotifyAccess = "all";
+          StartLimitBurst = "3";
+          TimeoutSec = 120;
+          WatchdogSec = 60;
+          RestartSec = 5;
+          Restart = "always";
+          LimitNOFILE = 1024;
```

Comment:
Please add a comment here, that this should be increased, in setups with many routing peers (e.g. BGP) and/or management connections.

Reply:
or even make a `services.frr` option out of this?

Reply:
I have introduced `services.frr.openFilesLimit`, seemed to be the most clean way.